mkosi

```diff
@@ -2909,6 +2909,64 @@ def setup_package_cache(args: CommandLineArguments) -> Optional[tempfile.Tempora
     return d
 
 
+class ArgumentConfigParser():
+    options = dict()
+    argparser = None
+    curr_group_title = None
+    curr_group = None
+    FLAGS = enum.Enum('FLAGS', 'Extend')
+
+    def __init__(self, *kargs, **kwargs):
+        self.argparser = argparse.ArgumentParser(*kargs, **kwargs)
+
+    def add_section(self, *kargs, **kwargs):
+        fp_name = kwargs.pop('file_parser_name', None)
```

Comment:
Please use the normal kwarg-only syntax:
```python3
def add_section(self, *args, file_parser_name=None, **kwargs):
    ...
```
The syntax with `pop` is just a work-around for python2 which we don't support anyway.

Also, I think the second args should traditionally be `args` not `kargs`.

Or is this interface somehow required by `argparse`?